_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,575 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][]: 
+ - where the attribute `key` is a [printable character][] key; and
+ - the attribute `isComposing` is set to `false`; or the [keyboard event][] is part of an [composition event][] whose attribute `data` includes at least one [non-printable character][] key; and
```

Comment:
Also, I am not sure that composition events is what you're looking for.
As far as I understand, composition events are more for stuff like the `compose` key (e.g. `<compose>` `<dead_acute>` `<e>` produce `é`, but each key is pressed separately).

This is supported by https://www.w3.org/TR/uievents/#events-composition-key-events which show single key pressed one by one (i.e. `keydown` is followed by `keyup`) rather than multiple keys pressed together (i.e. two or more `keydown` before we start the `keyup`).

So, I do not think that composition events are what is needed for modifiers (`Ctrl` and such). Although I do agree that if part of a composition event, things are probably OK but even if the composition event only contains printable character keys (e.g. a page reacting to the sequence `h` `e` `l` `l` `o` `w` `o` `r` `l` `d` but not to any of the individual keys is OK).

Reply:
I understand what you are pointing out, and i'll probably need to do a little bit more research about the composition events. But without them we can only target some modifier keys (those who are available in the [event interface](https://www.w3.org/TR/uievents/#events-keyboardevents)), which are only a subset of the non-printable characters keys. But now that i'm revisiting the spec, i think i can use the getModifierState() mehtod to know if any modifier keys are being used. I'll change to use that